sdk/resourcemanager/azure-resourcemanager-appservice/src/main/java/com/azure/resourcemanager/appservice/implementation/WebAppImpl.java

```diff
@@ -360,4 +369,47 @@ public Mono<KuduDeploymentResult> pushDeployAsync(DeployType type, File file, De
             return Mono.error(e);
         }
     }
+
+    @Override
+    public DeploymentBuildStatus getDeploymentStatus(String deploymentId) {
+        return getDeploymentStatusAsync(deploymentId).block();
+    }
+
+    @Override
+    public Mono<DeploymentBuildStatus> getDeploymentStatusAsync(String deploymentId) {
+        // "GET" LRO is not supported in azure-core
+        String deploymentStatusUrl = String.format(
+            "%s/subscriptions/%s/resourceGroups/%s/providers/Microsoft.Web/sites/%s/deploymentStatus/%s?api-version=%s",
+            AzureEnvironment.AZURE.getResourceManagerEndpoint(),
+            this.manager().subscriptionId(),
+            this.resourceGroupName(),
+            this.name(),
+            deploymentId,
+            this.manager().serviceClient().getApiVersion());
+        HttpRequest request = new HttpRequest(HttpMethod.GET, deploymentStatusUrl);
```

Comment:
You should call the `WithResponse` API instead of making the URL. The API is in 2022-03-01 swagger (that is the reason we supports it at this version).

Reply:
ok